server/sonar-web/src/main/js/apps/issues2/sidebar/Sidebar.js

```diff
@@ -19,20 +19,37 @@
  */
 // @flow
 import React from 'react';
-import TypeFacet from './TypeFacet';
+import AssigneeFacet from './AssigneeFacet';
+import AuthorFacet from './AuthorFacet';
+import DirectoryFacet from './DirectoryFacet';
+import FileFacet from './FileFacet';
+import LanguageFacet from './LanguageFacet';
+import ModuleFacet from './ModuleFacet';
+import ProjectFacet from './ProjectFacet';
 import ResolutionFacet from './ResolutionFacet';
 import RuleFacet from './RuleFacet';
 import SeverityFacet from './SeverityFacet';
 import StatusFacet from './StatusFacet';
-import type { Query, Facet } from '../utils';
+import TagFacet from './TagFacet';
+import TypeFacet from './TypeFacet';
+import type {
+  Query,
+  Facet,
+  ReferencedComponent,
+  ReferencedUser,
+  ReferencedLanguage
+} from '../utils';
```

Comment:
What do you think of creating a `js/types` folder with files containing all our re-usable types ? It would make it easier to know where to find them and check if a type already exists.